jobs/openshift/cluster/integreatly/openshift-cluster-integreatly-test.yaml

```diff
@@ -17,12 +17,181 @@
+                  jobParams = [
+                    [$class: 'StringParameterValue', name: 'REPOSITORY', value: QE_REPO_URL],
+                    [$class: 'StringParameterValue', name: 'BRANCH', value: 'master'],
+                    [$class: 'StringParameterValue', name: 'CLUSTER_URL', value: testOptions.openshiftMasterUrl],
+                    [$class: 'StringParameterValue', name: 'ADMIN_USERNAME', value: clusterAdminCredentials.clusterAdminUsername],
+                    [$class: 'StringParameterValue', name: 'ADMIN_PASSWORD', value: clusterAdminCredentials.clusterAdminPassword],
+                    [$class: 'StringParameterValue', name: 'NAMESPACE_PREFIX', value: 'openshift-'],
+                    [$class: 'StringParameterValue', name: 'CLEAN_RESOURCES', value: 'true']
+                  ]
+                  build job: jobName, parameters: jobParams
+                }
+              }
+            }
+
+            stage('Uninstall Integreatly') {
+              steps {
+                echo "[INFO] Uninstalling Integreatly from ${testOptions.clusterName}"
+                script {
+                  jobName = 'openshift-cluster-integreatly-uninstall'
+                  jobParams = [
+                    [$class: 'StringParameterValue', name: 'clusterName', value: testOptions.clusterName],
+                    [$class: 'StringParameterValue', name: 'openshiftMasterUrl', value: testOptions.openshiftMasterUrl],
+                    [$class: 'StringParameterValue', name: 'installationGitBranch', value: testOptions.gitUrl],
+                    [$class: 'StringParameterValue', name: 'installationGitBranch', value: testOptions.gitBranch],
+                    [$class: 'BooleanParameterValue', name: 'dryRun', value: params.dryRun]
+                  ]
+                  build job: jobName, parameters: jobParams
+                }
+              }
+            }
+          }
+
+          post {
+            always {
+              echo "[INFO] Deprovision ${testOptions.clusterName} cluster"
+              script {
+                if (testOptions.clusterName) {
+                  jobName = 'openshift-cluster-deprovision'
+                  jobParams = [
+                    [$class: 'StringParameterValue', name: 'clusterName', value: testOptions.clusterName],
+                    [$class: 'StringParameterValue', name: 'awsRegion', value: testOptions.awsRegion],
+                    [$class: 'StringParameterValue', name: 'awsAccountName', value: testOptions.awsAccountName],
+                    [$class: 'StringParameterValue', name: 'clusterDomainName', value: testOptions.domainName],
+                    [$class: 'BooleanParameterValue', name: 'dryRun', value: params.dryRun]
+                  ]
+                  build job: jobName, parameters: jobParams
+                } else {
+                  println "[INFO] Cluster name not available, skipping deprovision"
+                }
+              }
+            }
+          }
+        }
```

Comment:
Can you revert this and apply all changes in the Jenkinsfile https://github.com/integr8ly/ci-cd/blob/bf4c31b0566c47e04ffcf50f93f287ab09effa72/jobs/openshift/cluster/integreatly/test/Jenkinsfile. We don't put all the pipeline code in the job definition as it's harder to read and edit, and the jobs are eithe rpre processed to inject the dsl like this via the configure_script or loaded from Git.

Reply:
Ahh ok. I thought you prefer this. I'll apply the changes in the Jenkinsfile then.

Reply:
@mikenairn Updated.